File: Proyects/UniJava/TP4/ex4/Ellipse.java
public class Ellipse extends Shape implements Movable{
	private Point center;
	private double bigger, lower;
	
	public Ellipse(Point center, double bigger, double lower){
		this.center = center;
		this.bigger = bigger;
		this.lower = lower;
	}

	public Point getCenter(){
		return this.center;
	}

	public double perimeter(){
		return (Math.PI * (3 * (this.bigger/2 + this.lower/2) - Math.sqrt((3 * this.bigger/2 + this.lower/2) * (this.bigger/2 + 3 * this.lower/2))));
	}

	public double area(){
		return (Math.PI * this.bigger/2 * this.lower/2);
	}

	@Override
	public String toString(){
		String ret = "[Center{" + this.center + "},Bigger{" + this.bigger + "},lower{" + this.lower + "}]";
		return ret;
	}

	public void moveEast(double delta){
		this.center.moveEast(delta);
	}

	public void moveNorth(double delta){
		this.center.moveNorth(delta);
	}
}

File: Proyects/UniJava/TP8/ex4/ellipse.rb
require_relative 'shape'
class Ellipse < Shape
  def initialize(center, diameter_x, diameter_y)
    @center = center
    @diameter_x = diameter_x
    @diameter_y = diameter_y
  end

  def perimeter
    Math::PI / 2 * (@diameter_x + @diameter_y)
  end

  def area
    Math::PI / 4 * @diameter_x * @diameter_y
  end
end
